### PR TITLE
Assignment 2- Create PR for further Review

### PR DESCRIPTION
**Pull request by GoncaloJardim**

### CODE REVIEW

File: life_expectancy/cleaning.py [hunk 1]
```diff
@@ -5,23 +5,29 @@
 import pandas as pd
 import numpy as np
 
+
 DATA_PATH =  pathlib.Path(__file__).parent / 'data'
 
-def clean_data(region= None):
+def _load_data():
```
Comment (MariaStavr) [inline life_expectancy/cleaning.py]: As you well spotted, type hints are missing :)

File: life_expectancy/cleaning.py [hunk 2]
```diff
@@ -45,42 +51,54 @@ def clean_data(region= None):
...
+def main(region: str ="PT"):
+    """Pipeline with functions for:
+    - Loading data;
+    - Cleaning data;
+    - Saving data;"""
+
+    (
+        _load_data().
+        pipe(_clean_data, region).
+        pipe(_save_data)
```
Comment (MariaStavr) [inline life_expectancy/cleaning.py:90]: Thanks for the tip to use pipe method, I have updated my code with it.

**Review by MariaStavr — COMMENTED** [review]
Very good job! I didn't have much to add! :rocket:

**Review by MrCordeiro — COMMENTED** [review]
From my checklist, here's what I noticed:

**Are there comments?**

There could be more comments inside the _clean_data function. A chain of pandas commands can be difficult to traverse (like line 47). Meanwhile, other pandas functions, like in like 54 are easy to understand what they do, but the WHY (i.e why are you doing the at operation) is not. Comments can help with that.
 
**Are the variable names descriptive?**

There aren't many variables, but I can think of at least one instance where using an explanatory variable could help: on line 39 you use "life_expectancy.columns[1:]", but the reader will have no idea of what those columns refer to unless they open the data themselves. A nice alternative is to use the following explanatory variable:

```
year_cols = life_expectancy.columns[1:]
```

That way we know you're melting the year columns.

**Type hints**

The type hint that a function that doesn't return anything should get is `None` 😉

I loved the docstrings and the pipe design patterns feel like something @Gustavo-SF would be proud of!

**Comment by GoncaloJardim** [comment]
Changes made, based on Feedback from @MrCordeiro  and @MariaStavr

**Pull request merged**